src/SessionSetup/ConnectToTargetDialog.cpp

```diff
@@ -63,9 +79,17 @@ std::optional<TargetConfiguration> ConnectToTargetDialog::Exec() {
   int rc = QDialog::exec();
 
   if (rc != QDialog::Accepted) {
+    if (connection_metric_ != nullptr) {
+      connection_metric_->SetStatusCode(orbit_metrics_uploader::OrbitLogEvent::CANCELLED);
```

Comment:
This sets the status to cancelled in all rejected cases. This should only be set to cancelled if the user pressed the reject button. If the an error happened then this should be ::INTERNAL_ERROR.

Reply:
Also, if an error happened, the Error will be displayed to the user and the ScopedMetric will continue counting the time until the user dismisses the error. This is probably not what we want. There are several ways to solve that. The quickest is probably to release the connection_metric_ before the error is shown. The probably nicer  alternative would be to do the error displaying in main.cpp and change the Exec signature to `ErrorMessageOr<TargetConfiguration> Exec()`. In this case you could also remove connection_metric_ being a unique_ptr

Reply:
I am sorry, for some reason I did not see these lines of code.  My comment then must have seemed somewhat confusing. My bad!